pkg/container-collection/options.go

```diff
@@ -788,6 +799,63 @@ func WithOCIConfigEnrichment() ContainerCollectionOption {
 	}
 }
 
+// WithOCIConfigForInitialContainer enriches initial containers with the OCI config.
+// It assumes that the initial containers are already populated in the ContainerCollection
+func WithOCIConfigForInitialContainer() ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		for _, container := range cc.initialContainers {
+			ppid := 0
+			sPath := filepath.Join(host.HostProcFs, fmt.Sprintf("%d", container.ContainerPid()), "status")
+			sFile, err := os.Open(sPath)
+			if err != nil {
+				log.Errorf("OCIConfig enricher: failed to open status file for container %s: %s", container.Runtime.ContainerID, err)
+				continue
+			}
+			defer sFile.Close()
```

Comment:
If there are many containers in `cc.initialContainers`, the open file descriptors will accumulate and the defer will close all of them at the end.

I wonder if a util function is preferable to avoid this.